src/dawn/Optimizer/PassSetNonTempCaches.cpp

```diff
@@ -0,0 +1,322 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Optimizer/PassSetNonTempCaches.h"
+#include "dawn/Optimizer/Cache.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+#include "dawn/Optimizer/PassDataLocalityMetric.h"
+#include "dawn/Optimizer/PassSetCaches.h"
+#include "dawn/Optimizer/StatementAccessesPair.h"
+#include "dawn/Optimizer/StencilInstantiation.h"
+#include "dawn/SIR/ASTExpr.h"
+#include "dawn/Support/Unreachable.h"
+#include <iostream>
+#include <set>
+#include <vector>
+
+namespace dawn {
+
+struct AcessIDTolocalityMetric {
+  int accessID;
+  int dataLocalityGain;
+  bool operator<(const AcessIDTolocalityMetric& rhs) {
+    return dataLocalityGain < rhs.dataLocalityGain;
+  }
+};
+
+struct NameToImprovementMetric {
+  std::string name;
+  Cache cache;
+  int dataLocalityImprovement;
+};
+
+class GlobalFieldCacher {
+public:
+  /// @brief The GlobalFieldCacher class handles the caching for a given Multistage
```

Comment:
shouldnt this documentation be above the class declaration (rather than ctr)?

Reply:
The `@brief` statement yes, but not the `@param`. I think I wasn't entirely clear in my review on this.